sklearn/preprocessing/tests/test_common.py

```diff
@@ -53,3 +55,59 @@ def test_missing_value_handling(est):
                 X_test[:, [i]][~np.isnan(X_test[:, i])])
             assert_array_equal(Xt_col_nonan,
                                Xt_col[~np.isnan(Xt_col.squeeze())])
+
+
+@pytest.mark.parametrize(
+    "est",
+    [QuantileTransformer(n_quantiles=10, random_state=42)]
+)
+@pytest.mark.parametrize(
+    "func_sparse_format",
+    [sparse.csr_matrix,
+     sparse.csc_matrix]
+)
+def test_missing_value_handling_sparse(est, func_sparse_format):
+    # check that the preprocessing method let pass nan
+    rng = np.random.RandomState(42)
+    X = iris.data.copy()
+    n_missing = 50
+    X[rng.randint(X.shape[0], size=n_missing),
+      rng.randint(X.shape[1], size=n_missing)] = np.nan
+    X_train, X_test = train_test_split(X, random_state=1)
+    # sanity check
+    assert not np.all(np.isnan(X_train), axis=0).any()
+    assert np.any(np.isnan(X_train), axis=0).all()
+    assert np.any(np.isnan(X_test), axis=0).all()
+    X_test[:, 0] = np.nan  # make sure this boundary case is tested
+
+    # convert the array to a sparse array
+    X_train_sparse = func_sparse_format(X_train)
+    X_test_sparse = func_sparse_format(X_test)
+
+    Xt = est.fit(X_train_sparse).transform(X_test_sparse)
+    # missing values should still be missing, and only them
+    assert_array_equal(np.isnan(Xt.A), np.isnan(X_test))
+
+    # check that the inverse transform keep NaN
+    Xt_inv = est.inverse_transform(Xt)
+    assert_array_equal(np.isnan(Xt_inv.A), np.isnan(X_test))
+    # FIXME: we can introduce equal_nan=True in recent version of numpy.
+    # For the moment which just check that non-NaN values are almost equal.
+    assert_allclose(Xt_inv.A[~np.isnan(Xt_inv.A)], X_test[~np.isnan(X_test)])
+
+    for i in range(X.shape[1]):
+        # train only on non-NaN
+        col_train_sparse = func_sparse_format(
+            X_train[:, [i]][~np.isnan(X_train[:, i])])
```

Comment:
Maybe define this in a function,
```
def _get_valid_samples_by_column(X, i):
    """Get samples non NaN samples in column i"""
    return X_train[:, [i]][~np.isnan(X_train[:, i])])
```
and use here and in the `test_missing_value_handling_dense` function